src/js/common/utils/data-utils/omit.js

```diff
@@ -0,0 +1,13 @@
+import cloneDeep from './cloneDeep';
+
+/**
+ * Returns a deep clone of he provided object without the fields specified.
+ *
+ * @param {Array} fields
+ * @param {Object} object
+ */
+export default function omit(fields, object) {
+  const newObj = cloneDeep(object);
```

Comment:
We shouldn't deep clone this, it will break some assumptions we make about immutable data. You can use reduce to build a new object from the list of keys from the old one and skip the fields in `fields`.

Reply:
Using reduce makes a lot of sense.

Is the assumption you're talking about the bit where we say "data is immutable; don't change it?" So we shouldn't _need_ to deep clone this? My concern was with nested objects. If we had something like so:

```js
const objA = { a: { b: 'b' }, c: 'c' };
const objB = _.omit(['c'], objA);
objB.a.b = 'c';
console.log(objA); // { a: { b: 'c' }, c: 'c' }
```

I was trying to ensure the return result was completely separate from the input, so modifying the return result _cannot_ modify the input.

Reply:
In general, we make the assumption that we're treating data as immutable and not changing it directly. There are two advantages to that. One is that an object you have isn't going to be changed by some far away function. The other benefit is that if you have two objects, you can compare them by doing `===` and `!==`, instead of having to do a deep equality check. You're trying to enforce the first benefit at the expense of the second, because now `objB.b` and `objA.b` are deeply equal, but `objB.b !== objA.b`. The lodash version of omit doesn't work this way.

We don't generally try to enforce the first advantage in this way because it means we're doing lots of cloning without gaining much. That second advantage I mentioned is one that redux-connect uses to make updates more efficient and one we use in React components with shouldComponentUpdate to make performance improvements when necessary. So losing it can have more of an impact.

If we feel the need to enforce the assumption that you can't change objects directly, then we would probably want to use a library like ImmutableJS or seamless-immutable, but those have tradeoffs.